### PR TITLE
SetAlternate

### PR DESCRIPTION
**Pull request by burrbull**
With this PR passing `into_alternate()` is not necessary (but still can be used) when you work with `Serial`, `Spi`, `I2c` or `I2s`.
Pins will be internally switched in alternated mode and restored on `release`.

### CODE REVIEW

File: src/fmpi2c.rs [hunk 1]
```diff
@@ -64,25 +64,33 @@ where
...
+
+impl<SCL, SDA, const SCLA: u8, const SDAA: u8> FMPI2c<FMPI2C1, (SCL, SDA)>
 where
-    SCL: PinScl<FMPI2C1>,
-    SDA: PinSda<FMPI2C1>,
+    SCL: PinScl<FMPI2C1, A = Const<SCLA>> + SetAlternateOD<SCLA>,
+    SDA: PinSda<FMPI2C1, A = Const<SDAA>> + SetAlternateOD<SDAA>,
 {
-    pub fn new<M: Into<FmpMode>>(i2c: FMPI2C1, pins: (SCL, SDA), mode: M) -> Self {
+    pub fn new<M: Into<FmpMode>>(i2c: FMPI2C1, mut pins: (SCL, SDA), mode: M) -> Self {
```
Comment (therealprof) [inline src/fmpi2c.rs:93]: Since you've been mentioning breaking changes in the CHANGELOG, why is this not considered a breaking change?

Comment (burrbull) [reply src/fmpi2c.rs:93]: Are you asking about all PR? If so, I've not found it breaks something that had worked earlier.

If you are about  this `mut`. It is internal `mut`. It does not affect on API.

Comment (therealprof) [reply src/fmpi2c.rs:93]: Hm, it is a public method so someone could use it, no?

Comment (burrbull) [reply src/fmpi2c.rs:93]: This is not type `mut`. This is value `mut`. Inside function.

Comment (burrbull) [reply src/fmpi2c.rs:93]:
```rust
fn new(mut pins: (SCL, SDA)) {
    // do something with `pins`
}
```
is just shorthand for:
```rust
fn new(pins: (SCL, SDA)) {
    let mut pins = pins;
    // do something with `pins`
}
```

Comment (therealprof) [reply src/fmpi2c.rs:93]: 🤦‍♂️ Somehow I was seeing a reference there. Never mind.

**Review by therealprof — APPROVED** [review]
LGTM, thanks.

bors r+

**Review by therealprof — COMMENTED** [review]
bors r-

**Review by therealprof — APPROVED** [review]
Now looks good to me for real.

bors r+

**Pull request merged**